core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -322,29 +322,60 @@ public static Snapshot snapshotAfter(Table table, long snapshotId) {
   }
 
   /**
-   * Returns the ID of the most recent snapshot for the table as of the timestamp.
+   * Returns the ID of the most recent snapshot on the given branch as of the given time in
+   * milliseconds
    *
    * @param table a {@link Table}
+   * @param branch a {@link String}
    * @param timestampMillis the timestamp in millis since the Unix epoch
    * @return the snapshot ID
-   * @throws IllegalArgumentException when no snapshot is found in the table older than the
-   *     timestamp
+   * @throws IllegalArgumentException when no snapshot is found in the table, on the given branch
+   *     older than the timestamp
    */
-  public static long snapshotIdAsOfTime(Table table, long timestampMillis) {
```

Comment:
We cannot change the behavior of this method to use ancestors instead of table history.